### PR TITLE
feat: [PIE-7456]: refactor execution/pipeline filters, remove context that's unnecessarily used at other places like environmental details

### PR DESCRIPTION
**Pull request by udayvunnam**
### Summary

- ExecutionListFilterContext is unnecessary for reusing only the parts of the execution list at other places. Removed this. 
- Extracted ExecutionCompareYamlHeader to be used at other places like environmental details
- cleanup
- fixes https://harness.atlassian.net/browse/PIE-6093
- fixes https://harness.atlassian.net/browse/CDS-48309
- fixes https://harness.atlassian.net/browse/CDS-50530 in better way
- Still, there's duplication in PipelineListFiter and ExecutionListFilter which was existing. That can be extracted later.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/components/Filter/FilterSelector/FilterSelector.module.scss [hunk 1]
```diff
@@ -41,8 +37,8 @@
   width: 20px;
   line-height: 20px;
   border-radius: 50%;
-  background-color: #0a3364;
-  color: white;
+  background-color: var(--primary-9);
+  color: var(--white);
```
Comment (udayvunnam) [inline src/modules/10-common/components/Filter/FilterSelector/FilterSelector.module.scss:41]: use css: vars

File: src/modules/70-pipeline/pages/execution-list/ExecutionListFilter/ExecutionListFilter.tsx [hunk 2]
```diff
@@ -249,8 +262,8 @@ export function ExecutionListFilter(): React.ReactElement {
   }
 
   return (
-    <div className={css.executionFilter} onClick={() => dispatchCustomEvent(FORM_CLICK_EVENT, {})}>
-      <FilterSelector<FilterDTO>
```
Comment (udayvunnam) [inline src/modules/70-pipeline/pages/execution-list/ExecutionListFilter/ExecutionListFilter.tsx:253]: This was unnecessary, unused

**Comment by udayvunnam** [comment]
retrigger titlecheck

**Review by rameshx — APPROVED** [review]

**Comment by udayvunnam** [comment]
retrigger cypress-pipeline

File: src/modules/70-pipeline/components/ExecutionCompareYaml/ExecutionCompareYamlHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { Button, ButtonVariation, Layout, Page, Text } from '@harness/uicore'
+import React, { ReactElement } from 'react'
+import { useStrings } from 'framework/strings'
+import { useBooleanStatus } from '@common/hooks'
+import { useExecutionCompareContext } from './ExecutionCompareContext'
+import { ExecutionCompareYaml } from './ExecutionCompareYaml'
+
+export function ExecutionCompareYamlHeader(): ReactElement {
+  const { cancelCompareMode, compareItems } = useExecutionCompareContext()
+  const { state: showCompareExecutionDrawer, close, open } = useBooleanStatus(false)
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/components/ExecutionCompareYaml/ExecutionCompareYamlHeader.tsx:18]: Just curious. What is the benefit of doing this?

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/70-pipeline/components/ExecutionCompareYaml/ExecutionCompareYamlHeader.tsx (hunk 3 above)
Comment (udayvunnam) [reply src/modules/70-pipeline/components/ExecutionCompareYaml/ExecutionCompareYamlHeader.tsx:18]: Do you mean useBooleanStatus hook or ExecutionCompareYamlHeader component?

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**